torch/csrc/jit/script/compiler.cpp

```diff
@@ -722,9 +731,20 @@ struct to_ir {
         }
       }
       ensureTensors(return_stmt.range(), results);
+      std::vector<TypePtr> flattened_return_types;
+      if (def_and_types.return_type) {
+        if (def_and_types.return_type->kind() == TypeKind::TupleType) {
+          const auto &tuple_type_elmts = def_and_types.return_type->cast<TupleType>()->elements();
```

Comment:
`FunctionSchema` already stores return types pre-flattened and would remove the need to do it here. This is another reason to parse get_annotation into `FunctionSchema` first.